test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc

```diff
@@ -539,5 +550,35 @@ TEST_F(FileUtilsTest, ReadFileContents) {
   EXPECT_EQ(data, output);
 }
 
+TEST(QuicPlatformTest, PickUnsedPort) {
+  int port = QuicPickUnusedPortOrDie();
+  std::vector<Envoy::Network::Address::IpVersion> supported_versions =
+      Envoy::TestEnvironment::getIpVersionsForTest();
+  for (auto ip_version : supported_versions) {
+    Envoy::Network::Address::InstanceConstSharedPtr addr =
+        Envoy::Network::Test::getCanonicalLoopbackAddress(ip_version);
+    Envoy::Network::Address::InstanceConstSharedPtr addr_with_port =
+        Envoy::Network::Utility::getAddressWithPort(*addr, port);
+    Envoy::Network::IoHandlePtr io_handle =
+        addr_with_port->socket(Envoy::Network::Address::SocketType::Datagram);
+    // binding of given port should success.
+    EXPECT_EQ(0, addr_with_port->bind(io_handle->fd()).rc_);
+  }
+}
+
+TEST(QuicPlatformTest, FailToPickUnsedPort) {
+  Envoy::Api::MockOsSysCalls os_sys_calls;
+  Envoy::TestThreadsafeSingletonInjector<Envoy::Api::OsSysCallsImpl> os_calls(&os_sys_calls);
+  // Actually create sockets.
+  EXPECT_CALL(os_sys_calls, socket(_, _, _)).WillRepeatedly([](int domain, int type, int protocol) {
+    int fd = ::socket(domain, type, protocol);
+    return Envoy::Api::SysCallIntResult{fd, errno};
+  });
+  // Fail bind call's to mimic port exhaustion.
+  EXPECT_CALL(os_sys_calls, bind(_, _, _))
+      .WillRepeatedly(Return(Envoy::Api::SysCallIntResult{-1, EADDRINUSE}));
+  EXPECT_DEATH(QuicPickUnusedPortOrDie(), "Failed to pick a port for test.");
```

Comment:
In #6393 I've changed
1. QuicPlatformTest from TEST to TEST_F.
2. EXPECT_DEATH to EXPECT_DEATH_LOG_TO_STDERR.

Please sync and do the same for the new tests.

Reply:
done